tests/components/jewish_calendar/test_calendar.py

```diff
@@ -0,0 +1,335 @@
+            [CalendarEventType.HOLIDAY, CalendarEventType.DATE],
+        ),
+        (list(CalendarEventType), list(CalendarEventType)),
+        (["invalid_event", CalendarEventType.DATE], [CalendarEventType.DATE]),
+    ],
+)
+@pytest.mark.usefixtures("setup")
+async def test_calendar_event_configuration(
+    hass: HomeAssistant, expected_events: list[str], get_calendar_events
+) -> None:
+    """Test calendar event configuration is handled properly."""
+    state = hass.states.get(CALENDAR_ENTITY_ID)
+    assert state is not None
+
+    # Get events for a date range that should have various event types
+    start_date = dt.date(2024, 1, 12)  # Friday
+    end_date = dt.date(2024, 1, 14)  # Sunday
+    events = await get_calendar_events(hass, start_date, end_date)
+
+    # Filter out invalid event types from expected
+    valid_expected_events = {
+        event_type for event_type in expected_events if event_type in CalendarEventType
+    }
+
+    # Extract actual event types from descriptions
+    found_event_types = set()
+    for event in events:
+        description = event.get("description", "")
+        if description.startswith("Hebrew date"):
+            found_event_types.add(CalendarEventType.DATE)
+        elif description.startswith("Parshat Hashavua"):
+            found_event_types.add(CalendarEventType.WEEKLY_PORTION)
+        elif description.startswith("Jewish Holiday"):
+            found_event_types.add(CalendarEventType.HOLIDAY)
+        elif description.startswith("Sefirat HaOmer"):
+            found_event_types.add(CalendarEventType.OMER_COUNT)
+        elif description.startswith("Daf Yomi"):
+            found_event_types.add(CalendarEventType.DAF_YOMI)
+        elif description.startswith("Candle lighting time"):
+            found_event_types.add(CalendarEventType.CANDLE_LIGHTING)
+        elif description.startswith("Havdalah time"):
+            found_event_types.add(CalendarEventType.HAVDALAH)
+
+    # Should not have events from unconfigured types
+    unexpected_event_types = found_event_types - valid_expected_events
+    assert not unexpected_event_types, (
+        f"Found events for unconfigured types: {unexpected_event_types}"
+    )
+
+    # Should have at least some events
+    assert len(events) > 0
+
+
+@pytest.mark.parametrize("test_date", [dt.date(2024, 1, 15)])
+@pytest.mark.usefixtures("setup_at_time")
+async def test_get_events_date_range(
+    hass: HomeAssistant, test_date: dt.date, get_calendar_events
+) -> None:
+    """Test getting events for a date range."""
+    end_date = dt.date(2024, 1, 17)
+    events = await get_calendar_events(hass, test_date, end_date)
+    assert isinstance(events, list)
+    assert len(events) >= 3  # At least date events for 3 days
+
+    # All events should have required fields
+    for event in events:
+        assert "summary" in event
+        assert "description" in event
+        assert "start" in event
+        assert "end" in event
```

Comment:
This sanity check is good, but I think a snapshot of the events would be good too.